### DISCUSSION TITLE
Deprecate / remove support for UTF-16 / CHAR UNICODE

### DISCUSSION THREAD

**Discussion by fisharebest** [FamilySearch/GEDCOM#78] [Q&A] [answered]
To my knowledge, no application has created GEDCOM files using `1 CHAR UNICODE` /  `UTF-16`.

I have only encountered some hand-crafted test files.

Since `UTF-8` and `UNICODE` support was first added to the GEDCOM spec, `UTF-8` has emerged as a de-facto standard.

It is trivial to create a program/script to convert a `UNICODE` GEDCOM file into a `UTF-8` one.  (`UTF-16` is a strict subset of `UTF-8`).

I'd like to propose that we mark `UTF-16` as deprecated.  e.g.

> Support for encodings other than `UTF-8` are deprecated, and will be removed in 8.0.0

The only valid value for `HEAD.CHAR` is now `UTF-8`.  We can remove it.

This would help with #5

**Comment by Colinsp** [comment]
Family Historian V6 used UTF-16 for its Gedcom. This was replaced with
UTF-8 for Version 7 released earlier this year.

On Tue, 19 Oct 2021 at 09:34, Greg Roach ***@***.***> wrote:

> To my knowledge, no application has created GEDCOM files using 1 CHAR
> UNICODE / UTF-16.
>
> I have only encountered some hand-crafted test files.
>
> Since UTF-8 and UNICODE support was first added to the GEDCOM spec, UTF-8
> has emerged as a de-facto standard.
>
> It is trivial to create a program/script to convert a UNICODE GEDCOM file
> into a UTF-8 one. (UTF-16 is a strict subset of UTF-8).
>
> I'd like to propose that we mark UTF-16 as deprecated. e.g.
>
> Support for encodings other than UTF-8 are deprecated, and will be
> removed in 8.0.0
>
> The only valid value for HEAD.CHAR is now UTF-8. We can remove it.
>
> This would help with #5 <https://github.com/FamilySearch/GEDCOM/issues/5>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FamilySearch/GEDCOM/issues/76>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPVESNFLKQB4HFWTTKKU6TUHUUQ5ANCNFSM5GIRJSGA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Regards

Colin

www.turnspain.com
www.spencer-genealogy.com
https://le-fever.org

**Comment by tychonievich** [comment, marked as answer]
For context:
- FamilySearch GEDCOM 7.0 is UTF-8 only.
- We are not planning to edit the content of any previous GEDCOM specifications
- But we are discussing how to specify (roughly) "all GEDCOM versions ever" as part of [the media type specification](https://github.com/FamilySearch/GEDCOM/blob/main/version-detection/version-detection.md). Whether deprecation is the right tool for that document or not is not clear to me.
